OCPBUGS#55065: Updated the JSON examples in compatibility-with-multi-… (#92259)
openshift/openshift-docs: modules/configuring-localnet-switched-topology.adoc

You must map an additional network to the OVN bridge to use it as an OVN-Kubernetes additional network. Bridge mappings allow network traffic to reach the physical network. A bridge mapping associates a physical network name, also known as an interface label, to a bridge created with Open vSwitch (OVS).

You can create an `NodeNetworkConfigurationPolicy` (NNCP) object, part of the `nmstate.io/v1` API group, to declaratively create the mapping. This API is provided by the NMState Operator. By using this API you can apply the bridge mapping to nodes that match your specified `nodeSelector` expression, such as `node-role.kubernetes.io/worker: ''`. With this declarative approach, the NMState Operator applies additional network configuration to all nodes specified by the node selector automatically and transparently.

.Example NNCP object that maps two physical network interfaces to the network
[source,yaml]
----
apiVersion: nmstate.io/v1
kind: NodeNetworkConfigurationPolicy
metadata:
  name: bridge-mapping 
spec:
  nodeSelector:
    node-role.kubernetes.io/worker: '' 
  desiredState:
    ovn:
      bridge-mappings:
      - localnet: localnet1
        bridge: br-ex 

>       - localnet: localnet2

Reviewer: @dfitzmau Could you remove this one as I see in preview of entire doc page that you do have NNCP with secondary interface the creates localnet2. Thanks!

Author: Thanks. Removed.

// tag::localnet-content[]
You must map an additional network to the OVN bridge to use it as an OVN-Kubernetes additional network. Bridge mappings allow network traffic to reach the physical network. A bridge mapping associates a physical network name, also known as an interface label, to a bridge created with Open vSwitch (OVS).

> You can create an `NodeNetworkConfigurationPolicy` object, part of the `nmstate.io/v1` API group, to declaratively create the mapping. This API is provided by the NMState Operator. By using this API you can apply the bridge mapping to nodes that match your specified `nodeSelector` expression, such as `node-role.kubernetes.io/worker: ''`. With this declarative approach, the NMState Operator applies additional network configuration to all nodes specified by the node selector automatically and transparently.

Reviewer: @dfitzmau how about add the NNCP example as well to map the two physical network names?
<pre>
apiVersion: nmstate.io/v1
kind: NodeNetworkConfigurationPolicy
metadata:
  name: bridge-mapping 
spec:
  nodeSelector:
    node-role.kubernetes.io/worker: '' 
  desiredState:
    ovn:
      bridge-mappings:
      - localnet: localnet1
        bridge: br-ex 
     - localnet: localnet2
        state: present 
</pre>

Author: :heavy_check_mark:

> You can create an `NodeNetworkConfigurationPolicy` object, part of the `nmstate.io/v1` API group, to declaratively create the mapping. This API is provided by the NMState Operator. By using this API you can apply the bridge mapping to nodes that match your specified `nodeSelector` expression, such as `node-role.kubernetes.io/worker: ''`. This declarative approach is recommended because the NMState Operator applies additional network configuration to all nodes specified by the node selector automatically and transparently.

Reviewer 2: Can we rework this to avoid the word "recommended"? IBMSG suggests staying away from that word in various contexts; in general, I try to avoid it, but sometimes it is acceptable: https://www.ibm.com/docs/en/ibm-style?topic=information-claims-recommendations

Maybe something like:

```suggestion
You can create an `NodeNetworkConfigurationPolicy` object, part of the `nmstate.io/v1` API group, to declaratively create the mapping. This API is provided by the NMState Operator. By using this API you can apply the bridge mapping to nodes that match your specified `nodeSelector` expression, such as `node-role.kubernetes.io/worker: ''`. This declarative approach allows the NMState Operator to automatically and transparently apply additional network configuration to all nodes specified by the node selector.
```

but maybe you have a better way.

Author: :heavy_check_mark:

<3> The name for the additional network from which traffic is forwarded to the OVS bridge. This additional network must match the name of the `spec.config.name` field of the `NetworkAttachmentDefinition` CRD that defines the OVN-Kubernetes additional network.
<4> The name of the OVS bridge on the node. This value is required only if you specify `state: present`.
<5> The state for the mapping. Must be either `present` to add the bridge or `absent` to remove the bridge. The default value is `present`.

> +

Reviewer: With 4.19, we set up CUDN that configures the NAD as described by JSON below? Should CUDN and NAD relation highlighted here?

Reviewer: Let us stick with NAD only here as I think Joe is working on CUDN with localnet topology.

> The following JSON example configures a localnet, `localnet1`, secondary network:

Reviewer 2: Consider (though I don't have a preference):

```suggestion
The following JSON example configures a localnet secondary network named `localnet1`.
```

Author: :heavy_check_mark:

+
[source,json]
----
{
  "cniVersion": "0.3.1",
  "name": "ns1-localnet-network",
  "type": "ovn-k8s-cni-overlay",

>   "topology":"localnet1",

Reviewer 3: Here it should say: `topology: localnet`

<5> The name for the additional network from which traffic is forwarded to the OVS bridge. This additional network must match the name of the `spec.config.name` field of the `NetworkAttachmentDefinition` CRD that defines the OVN-Kubernetes additional network.
<6> The name of the OVS bridge on the node. This value is required only if you specify `state: present`.
<7> The state for the mapping. Must be either `present` to add the bridge or `absent` to remove the bridge. The default value is `present`.
+

> The following JSON example configures a localnet, `localnet2`, secondary network:

Reviewer 2: Same consideration:

```suggestion
The following JSON example configures a localnet secondary network named `localnet2`.
```

Author: :heavy_check_mark:

+
[source,json]
----
{
  "cniVersion": "0.3.1",
  "name": "ns1-localnet-network",
  "type": "ovn-k8s-cni-overlay",

>   "topology":"localnet2",

Reviewer 3: Here it should say: `topology: localnet`

Author: :heavy_check_mark:
